### PR TITLE
Indicator fixes and update MDC Ripple

### PR DESCRIPTION
**Pull request by danielraggs**
Fixes various bugs with the MDC Tab Bar and its navigation behavior:

- **Tabs as links**: Indicator will now respond to changes in the URL or query params not triggered by clicking the tabs directly.
- When the array of tabs changes to a different set of tabs without un-rendering the parent tab bar, the indicator is updated accordingly.
- Tab bar scroller will now scroll hidden active tabs into view.

Additionally, MDC Ripple has been upgraded to the latest version to fix styling issues in Edge.

### CODE REVIEW

**Comment by Kerrick** [comment]
@danielraggs Now that MDC Tabs has shipped 0.3.0, is this PR ready for review + merge?

**Comment by danielraggs** [comment]
@Kerrick This is ready for code review and to be merged!

File: addon/components/mdc-tab-bar-scroller.js [hunk 1]
```diff
@@ -144,6 +144,16 @@ export default Ember.Component.extend(MDCComponent, {
     },
     deregisterTabBar() {
       set(this, 'tab-bar', null);
+    },
+    scrollActiveTabIntoView(index) {
+      // TODO: Need to submit an issue asking Google to make a scroll method that only triggers if tab is hidden, must do this until then
```
Comment (Kerrick) [inline addon/components/mdc-tab-bar-scroller.js]: Is there an issue for this?

File: addon/components/mdc-tab-bar.js [hunk 2]
```diff
@@ -110,7 +123,12 @@ export default Ember.Component.extend(MDCComponent, {
   },
   setTabActiveAtIndex(index, isActive) {
     if (get(this, 'links')) {
-      this.tabAt(index)._invoke({ stopPropagation() {}, preventDefault() {} }); // TODO: Probably shouldn't be calling private APIs or stubbing events
+      if (this.tabAt(index) && isActive) {
+        this.tabAt(index)._invoke({ stopPropagation() {}, preventDefault() {} }); // TODO: Probably shouldn't be calling private APIs or stubbing events
```
Comment (Kerrick) [inline addon/components/mdc-tab-bar.js]: Another TODO

Comment (danielraggs) [reply addon/components/mdc-tab-bar.js]: This one was added by you I think?

File: addon/components/mdc-tab-bar-scroller.js (hunk 1 above)
Comment (danielraggs) [reply addon/components/mdc-tab-bar-scroller.js]: Not at the moment.

File: addon/components/mdc-tab-bar.js (hunk 2 above)
Comment (Kerrick) [reply addon/components/mdc-tab-bar.js]: Yep I didn't notice that it was moved, not added.

**Comment by Kerrick** [comment]
Thank you so much for the PR! This will ship out with v0.0.37 👍

**Comment by MrChriZ** [comment]
I've just tried creating an Ember app with Ember Material Components Web 0.0.38.
I added this to the application.hbs:

> 
{{#mdc-tab-bar as |bar|}}
  {{#bar.tab "fruits" active=isFirstActive become-active=(action (mut isFirstActive))}}
    Fruits
  {{/bar.tab}}
  {{#bar.tab "vegetables" active=isSecondActive become-active=(action (mut isSecondActive))}}
    Vegetables
  {{/bar.tab}}
  {{#bar.tab "meats" active=isThirdActive become-active=(action (mut isThirdActive))}}
    Meats
  {{/bar.tab}}
{{/mdc-tab-bar}}

And created three routes to go with it.

When changing route using the address bar the indicator was not following the route?
Am I missing anything?

**Comment by Kerrick** [comment]
@MrChriZ You shouldn't pass `active` and `become-active` to the `{{bar.tab}}` components if you want their active state to be tied to the router. Try it like this:

https://github.com/secondstreet/ember-material-components-web/blob/da299ef5f0a5f1f4c0bba0c016cadfb7fa459636/tests/dummy/app/templates/application.hbs#L838-L848

**Comment by MrChriZ** [comment]
Ah that works thanks :)

**Pull request merged**